### PR TITLE
Update bs-platform to 3.1.5 + dependencies

### PR DESCRIPTION
**Pull request by jchavarri**
Updates bs-platform to v3.1.5 and some other deps:

- `bs-json` to 2.0.0
- `reason-react` to 0.4.2
- `vroom` to latest

This PR still doesn't compile as is, it depends on https://github.com/glennsl/bs-refmt/pull/2 and https://github.com/glennsl/bs-in-a-box/pull/2 to be merged, so the commits in this one can be updated.

The only relevant change in code is in `App.re` due to `didMount` changing its signature.

### CODE REVIEW

File: src/view/App.re [hunk 1]
```diff
@@ -29,13 +30,8 @@ let make = (~data: Store.state(Store.data),
...
 
-  didMount: ({ send, state }) => {
-    ReasonReact.Update({
-      ...state,
-      worker: ref(Some(Worker.make(
-        ~onMessage  = {message => send(WorkerMessage(message))}
-      )))
-    })
+  didMount: ({ send }) => {
+    send(Init(message => send(WorkerMessage(message))))
```
Comment (glennsl) [inline src/view/App.re:34]: Is it necessary to pass the callback in here? I don't see what this closes over that isn't also available in the reducer?

**Review by glennsl — COMMENTED** [review]
Awesome, thank you! The other PRs have been merged, but maybe you'd want to update `bs.js` too. Let me know when it's ready to merge.

File: src/view/App.re (hunk 1 above)
Comment (jchavarri) [reply src/view/App.re:34]: The `send` function? Afaik in the reducer you only have access to `action` and `state`... but maybe I'm missing something?

**Comment by jchavarri** [comment]
I've [updated](https://github.com/rebench/rebench.github.io/pull/6/commits/325ce7c283dbb1981eae3ff7fd8ad2cc6e24fcb3) the dependencies to point `bs-refmt` and `bs-in-a-box` to the latest commits in their `master` branches.

Unless the callback in the `App` reducer needs more massaging, this should be ready to merge. `bs.js` would be nice but every time I try to compile Bucklescript with jsoo i end up spending a few hours in it 😁

File: src/view/App.re (hunk 1 above)
Comment (glennsl) [reply src/view/App.re:34]: Right, you'd have to use `SideEffects` to get `send` and then do another round through the reducer to update it, so definitely better the way you have it now.

Integrating RR components with external services really leaves something to be desired... Either that or I'm thinking about this all wrong...

**Comment by glennsl** [comment]
> every time I try to compile Bucklescript with jsoo i end up spending a few hours in it 

lol, yeah, that's just the rite of passage I think ;) Could probably just steal it from the Reason or BuckleScript playground though, I think they've been kept decently up to date.

Anyway, merging this now and will worry about that later. Thanks for this! :)

**Pull request merged**